tests/console/yast2_kdump.pm

```diff
@@ -0,0 +1,61 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Test scenario which configures Kdump with a YaST module
+# and checks configuration without rebooting the system.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base "y2_module_consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use registration;
+use kdump_utils;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+    select_console('root-console');
+    my @conf_files = @{get_test_suite_data()->{kdump}};
+
+    # install kdump by adding additional modules
+    add_suseconnect_product('sle-module-desktop-applications');
+    add_suseconnect_product('sle-module-development-tools');
+    zypper_call('in yast2-kdump');
+
+    # Kdump configuration with YaST module
+    kdump_utils::activate_kdump;
+
+    # check service (without restarting)
+    systemctl('is-enabled kdump');
+
+    # check configuration files
+    for my $conf_file (@conf_files) {
+        my $path        = $conf_file->{path};
+        my $conf_output = script_output("cat $path");
+        for my $setting (keys %{$conf_file->{settings}}) {
+            my ($conf_line) = grep { /$setting=/ } split(/\n/, $conf_output);
+            die "Setting '$setting' not found in $path" unless $conf_line;
+            my $value = $conf_file->{settings}->{$setting};
+            assert_matches(qr/^$setting=["]?$value["]?$/, $conf_line,
+                "Found unexpected setting value in $path.");
```

Comment:
How about "Found unexpected value in $path"?

Reply:
we are using 'assert_matches', so it will tell you later on what doesn't match, let me show you how the error looks [here](https://openqa.suse.de/tests/5145453#step/yast2_kdump/28).
I didn't include this code in a library, I'm not sure how generic could be, the idea behind is:
- only get the text output once with openQA command, not several times
- filter with Perl the line in particular(the first occurrence)
- match a regex against it. Instead of matching exact value, which would create many test data file, I gave that flexibility, so those extra space or very small differences can be expressed with regex, for example /'.*A\d+'/ or /(A|B)/. That also could help if the setting is not sanitized as you mentioned.

For example I needed a regex to express that the crashkernel is added and is equal to a number expressed in Megabyes or that the path for the crash sometimes appears like `file:///var/crash` and sometimes like `/var/crash`.